client/my-sites/plan-features-2023-grid/features.tsx

```diff
@@ -13,29 +44,38 @@ const PlanFeatures2023GridFeatures: React.FC< {
 	return (
 		<>
 			{ features.map( ( currentFeature, featureIndex ) => {
+				const key = `${ currentFeature.getSlug() }-${ featureIndex }`;
+
+				const isFreePlanAndCustomDomainFeature =
```

Comment:
Non-blocker: this condition means that the each feature no longer binds to one single component using the same logic. By putting this in here, it means that all the feature item will have to go through this computation while most of them doesn't need it.

We don't have to address it now, but we should look for a better way in the future :)

Reply:
> By putting this in here, it means that all the feature item will have to go through this computation while most of them doesn't need it.

Can you please explain this a bit more? Do you mean this computation?
`const isFreePlanAndCustomDomainFeature = currentFeature.getSlug() === FEATURE_CUSTOM_DOMAIN && isFreePlan( planName );`

Reply:
Yes, it's done inside `features.map()`. Thus, when the anonymous function is being called on each item in `features`, it will be computed even if we know that only the custom domain feature item needs it. 

Fwiw this is where an object-oriented language would solve by subclassing. e.g. each item is a subclass of the parent class `FeatureItem` with an abstract method `renderComponent()`. However, I'm not quite sure what's the idiomatic React counterpart of this concept :)